### ISSUE TITLE
Select field examples have accessibility error 

### ISSUE THREAD

**Issue by Blackbaud-DiHuynh** [blackbaud/skyux-select-field#91]
### Expected behavior
Examples on official documentation are accessible. 

### Actual behavior
The examples failed this a11y rule
```
aXe - [Rule: 'label-content-name-mismatch'] Elements must have their visible text as part of their accessible name - WCAG: wcag253
       Get help at: https://dequeuniversity.com/rules/axe/3.5/label-content-name-mismatch?application=axeAPI
```

### Steps to reproduce
Open https://developer.blackbaud.com/skyux/components/select-field and run an a11y tool. 

### Severity
Low

### Impact
Confusing developer experience

[AB#1600307](https://blackbaud.visualstudio.com/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/1600307)

**Comment by Blackbaud-DiHuynh** [comment]
a slack conversation with Paul Crowder suggests that `aria-describedby` is a better attribute for button labels
